cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorSecretEventSource.java

```diff
@@ -47,7 +52,11 @@ protected void onEventReceived(Action action, Secret resource) {
             resource.getMetadata().getName(),
             resource.getMetadata().getResourceVersion());
 
-        getEventHandler().handleEvent(
-            new ConnectorSecretEvent(this, resource));
+        List<OwnerReference> ownerReferences = resource.getMetadata().getOwnerReferences();
+        if (!ownerReferences.isEmpty()) {
+            String connectorName = ownerReferences.get(0).getName();
+            getEventHandler().handleEvent(new Event(new ResourceID(connectorName, namespace)));
```

Comment:
I think we can use the namespace from the owner reference assuming that this kind of secrets are always on the same namespace as the related managed connector (the owner)